libs/datasets/sources/nytimes_dataset.py

```diff
@@ -1,36 +1,72 @@
 from libs.datasets import data_source
 from libs.datasets import dataset_utils
+from libs.datasets.dataset_utils import AggregationLevel
 from libs.datasets.common_fields import CommonIndexFields
 from libs.us_state_abbrev import ABBREV_US_UNKNOWN_COUNTY_FIPS
 
-# On 2020-07-24, CT reported a backfill of 440 additional positive cases.
-# https://portal.ct.gov/Office-of-the-Governor/News/Press-Releases/2020/07-2020/Governor-Lamont-Coronavirus-Update-July-24
-# See https://trello.com/c/rPiM6UF4/333-adjust-ct-case-data-based-on-reported-backfill-vs-current-cases
-# for more information on backfill.
-ESTIMATED_REMOVED_CT_CASES_COUNTY = {
-    "09001": 188,
-    "09003": 154,
-    "09005": 24,
-    "09007": 4,
-    "09009": 39,
-    "09011": 8,
-    "09013": 22,
-    "09015": 2,
-}
-
-
-def _remove_ct_backfill_cases(data: pd.DataFrame) -> pd.DataFrame:
-    # July 24th was the day there was a backfill of cases
-    is_on_or_after_july_24 = data[CommonFields.DATE] >= "2020-07-24"
-
-    # Remove backfilled tests from cumulatives on and after July 24th.
-    for fips, count in ESTIMATED_REMOVED_CT_CASES_COUNTY.items():
-        is_fips_data_after_jul_24 = is_on_or_after_july_24 & (data[CommonFields.FIPS] == fips)
-        data.loc[is_fips_data_after_jul_24, CommonFields.CASES] -= count
+BACKFILLED_CASES = [
+    # On 2020-07-24, CT reported a backfill of 440 additional positive cases.
+    # https://portal.ct.gov/Office-of-the-Governor/News/Press-Releases/2020/07-2020/Governor-Lamont-Coronavirus-Update-July-24
+    ("09", "2020-07-24", 440),
+    # https://portal.ct.gov/Office-of-the-Governor/News/Press-Releases/2020/07-2020/Governor-Lamont-Coronavirus-Update-July-29
+    ("09", "2020-07-29", 384),
+]
+
+
+def _calculate_county_adjustments(
+    data: pd.DataFrame, date: str, backfilled_cases: int, state_fips: str
+) -> Dict[str, int]:
+    """Calculating number of cases to remove per county, weighted on number of new cases per county.
+
+    Weighting on number of new cases per county gives a reasonable measure of where the backfilled
+    cases ended up.
+
+    Args:
+        data: Input Data.
+        date: Date of backfill.
+        backfilled_cases: Number of backfilled cases.
+        state_fips: FIPS code for state.
+
+    Returns: Dictionary of estimated fips -> backfilled cases.
+    """
+    is_county_level = data[CommonFields.AGGREGATE_LEVEL] == AggregationLevel.COUNTY
+    is_state = data[CommonFields.FIPS].str.match(f"{state_fips}[0-9][0-9][0-9]")
+    is_not_unknown = data[CommonFields.FIPS] != f"{state_fips}999"
+
+    fields = [CommonFields.DATE, CommonFields.FIPS, CommonFields.CASES]
+    cases = (
+        data.loc[is_state & is_not_unknown, fields]
+        .set_index([CommonFields.FIPS, CommonFields.DATE])
+        .sort_index()
+    )
+    cases = cases.diff().reset_index(level=1)
+    cases_on_date = cases[cases.date == date]["cases"]
+    return (cases_on_date / cases_on_date.sum() * backfilled_cases).round().to_dict()
```

Comment:
do you think it is possible for the new sum of cases in the counties to be more than 1 from the old sum of cases - backfilled if the rounding gets unlucky and a bunch of counties have a new case count that gets rounded in the same direction? Don't need to fix but maybe add a comment if it seems possible to you.

Reply:
It's definitely possible (especially for a state with more counties).  I'll add a comment!